### ISSUE TITLE
Exclude ZLogger functions from stack trace

### ISSUE THREAD

**Issue by FrankNine** [Cysharp/ZLogger#56]
Hello,

We are surveying ZLogger as a replacement of our Unity logger. The string formatting GC allocation is indeed smaller, but the GC allocation for stack trace is larger since `FormatLogState` has many different overloads and all of them are printed to the stack trace. I wonder if there is a way or workaround to exclude ZLogger functions from Unity stack trace? I tried tagging ZLogger functions with [DebuggerHiddenAttribute](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debuggerhiddenattribute) but it doesn’t work.

Thank you.

```
UnityEngine.Debug:Log (object)
ZLogger.Providers.UnityDebugLogProcessor:Post (ZLogger.IZLoggerEntry) (at Packages/com.coresdk.zlogger@1.6.1/Unity/ZLoggerUnityLoggerProvider.cs:56)
ZLogger.AsyncProcessZLogger:Log<ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>> (Microsoft.Extensions.Logging.LogLevel,Microsoft.Extensions.Logging.EventId,ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>,System.Exception,System.Func`3<ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>, System.Exception, string>) (at Packages/com.coresdk.zlogger@1.6.1/AsyncProcessZLogger.cs:30)
Microsoft.Extensions.Logging.Logger:<Log>g__LoggerLog|12_0<ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>> (Microsoft.Extensions.Logging.LogLevel,Microsoft.Extensions.Logging.EventId,Microsoft.Extensions.Logging.ILogger,System.Exception,System.Func`3<ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>, System.Exception, string>,System.Collections.Generic.List`1<System.Exception>&,ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>&)
Microsoft.Extensions.Logging.Logger:Log<ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>> (Microsoft.Extensions.Logging.LogLevel,Microsoft.Extensions.Logging.EventId,ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>,System.Exception,System.Func`3<ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>, System.Exception, string>)
Microsoft.Extensions.Logging.Logger`1<Logger>:Microsoft.Extensions.Logging.ILogger.Log<ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>> (Microsoft.Extensions.Logging.LogLevel,Microsoft.Extensions.Logging.EventId,ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>,System.Exception,System.Func`3<ZLogger.Entries.FormatLogState`7<object, int, int, int, int, int, int>, System.Exception, string>)
ZLogger.ZLoggerExtensions:ZLog<int, int, int, int, int, int> (Microsoft.Extensions.Logging.ILogger,Microsoft.Extensions.Logging.LogLevel,Microsoft.Extensions.Logging.EventId,System.Exception,string,int,int,int,int,int,int) (at Packages/com.coresdk.zlogger@1.6.1/ZLoggerExtensions.cs:1463)
ZLogger.ZLoggerExtensions:ZLogDebug<int, int, int, int, int, int> (Microsoft.Extensions.Logging.ILogger,string,int,int,int,int,int,int) (at Packages/com.coresdk.zlogger@1.6.1/ZLoggerExtensions.cs:1534)
Logger:ZLog () (at Assets/Logger.cs:46)
Logger:Update () (at Assets/Logger.cs:35)
```

**Comment by amelkor** [comment]
Bump
Are there any suggestions how to workaround this?

**Comment by tylearymf** [comment]
> Came across this one https://docs.unity3d.com/2022.2/Documentation/ScriptReference/HideInCallstackAttribute.html
Sadly, it's not included into LTS versions, only available from 2022.2

HideInCallstack does not work for generic methods. https://issuetracker.unity3d.com/issues/generic-methods-with-hideincallstack-are-present-when-strip-logging-callstack-option-is-enabled-in-the-console-window

**Comment by tylearymf** [comment]
With this repo, you can filter ZLogger logs without needing to recompile the DLL or add attributes to the source code manually. https://github.com/tylearymf/UnityCallstackFilter
